skimage/restoration/_denoise.py

```diff
@@ -355,3 +357,139 @@ def denoise_tv_chambolle(im, weight=50, eps=2.e-4, n_iter_max=200,
         raise ValueError('only 2-d and 3-d images may be denoised with this '
                          'function')
     return out
+
+
+def _wavelet_threshold(im, wavelet, threshold=None, noise_std=None, mode='soft'):
+    """Performs wavelet denoising.
+
+    Parameters
+    ----------
+    im : ndarray (2d or 3d) of ints, uints or floats
+        Input data to be denoised. `im` can be of any numeric type,
+        but it is cast into an ndarray of floats for the computation
+        of the denoised image.
+    wavelet : string
+        The type of wavelet to perform. Can be any of the options
+        [pywt.wavelist]_ outputs. For example, this may be any of ``{db1, db2,
+        db3, db4, haar}``.
+    noise_std : float, optional
+        The noise is estimated when noise_std is None (the default). It does
+        this as mentioned in pull request #1837.
```

Comment:
Do we want to keep a reference to a PR in a docstring (even for a hiden function)? I think the information can be summarized in a Notes section.